apps/buildog/src/app/organizations/[organizationId]/layout.tsx

```diff
@@ -8,7 +14,31 @@ export default function Layout({
   children: React.ReactNode;
   params: { organizationId: string };
 }) {
+  const { user } = useAuth();
+  const [organizations, setOrganizations] = useState([]);
   const { organizationId } = params;
+  const router = useRouter();
+
+  const fetchOrganizations = useCallback(async () => {
+    const response = await Service.makeAuthenticatedRequest("organizations");
+    if (response) {
+      setOrganizations(response);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!user) return;
+    fetchOrganizations();
+  }, [user, fetchOrganizations]);
+
+  useEffect(() => {
+    if (organizations.length === 0) return;
+
+    const orgExists = organizations.some((org) => org.OrganizationId === organizationId);
+    if (!orgExists) {
+      router.replace("/not-found");
+    }
+  }, [organizations, organizationId, router]);
```

Comment:
Why we need to request organizations? Do we need all of them here? The organization api have to return response if it exist or not. If exist you need to show the information to the user, if not exist you have to navigate user to /not-found page

Reply:
You wrote this in page.tsx, we need the improve structure here. I think we don't have to call this this request here